### PR TITLE
Add e2e tests 

### PR DESCRIPTION
**Pull request by bayandin**
This PR adds e2e tests for autoscaling. Some scripts from `neonvm` are used for environment preparation, so I've added `neonvm` as a submodule (for the time being, until we properly merge these repos).

Test `tests/e2e/autoscaling`:
- Create VM (1cpu, 1Gi)
- Start workload
- Check VM upscaled (to max 4cpu 4Gi)
- Stop workload
- Check VM downscaled (back to 1cpu 1Gi)

### CODE REVIEW

File: .github/workflows/test.yaml
Comment (tychoish) [inline .github/workflows/test.yaml:33]: my inclination, and this is somewhat counter to the example in the cloud repo, is to keep different suites/classes of tests in different workflow files just to make things easier to find/trace, particularly if there aren't dependencies between them.

File: .github/workflows/e2e-test.yaml
Comment (tychoish) [inline .github/workflows/e2e-test.yaml:24]: I'd love it if (more) of these were runnable outside of CI just to be able to verify changes locally before pushing to CI, or to use in development. (it'd be fine if this were linux only,)

File: Makefile
Comment (tychoish) [inline Makefile:3]: ```make -C neonvm kernel``` does this without the `cd`

similarly it'd be fun if `neonvm` were a target here that did submodule init or something related.

File: Makefile
Comment (tychoish) [inline Makefile:44]: I'd put all the variables near the top, just for readability (make always has  to parse/run global stuff anyway, so you don't get much by putting things at the bottom.

File: Makefile
Comment (tychoish) [inline Makefile:22]: nit: I'd maybe make all the arguments long form ...

**Review by tychoish — COMMENTED** [review]
this is great! I'm psyched.

File: .github/workflows/e2e-test.yaml
Comment (bayandin) [reply .github/workflows/e2e-test.yaml:24]: I've added a rough section to the README with instructions on how to run tests (and listing dependencies).

I plan to work more on Makefile (like conversing script to make targets for deduplication) and then will have a command to install/check dependencies.

File: Makefile
Comment (bayandin) [reply Makefile]: > `make -C neonvm kernel` does this without the `cd`

Nice! Reusing this.

> similarly it'd be fun if neonvm were a target here that did submodule init or something related.

Will do something like this in the following work with Makefile

**Comment by bayandin** [comment]
Ok, the PR is more or less ready for review.
It also includes a removal of local docker registry. I went through the setup mentioned in README, and it still works.

File: .github/workflows/e2e-test.yaml
Comment (tychoish) [reply .github/workflows/e2e-test.yaml:26]: perfect!

**Review by tychoish — APPROVED** [review]

File: Makefile
Comment (sharnoff) [inline Makefile:28]:
```suggestion
```

Don't need to load the VM informant image. The image is only created to store the VM informant binary before building the test VM.

File: Makefile
Comment (sharnoff) [inline Makefile:22]: not a blocker, but it would be good to actually include the git info here. If that seems like a pain to sort out, I'm happy to take care of it after merging

File: scripts/cluster-init.sh
Comment (sharnoff) [inline scripts/cluster-init.sh:30]:
```suggestion
for img in kube-autoscale-scheduler:dev autoscaler-agent:dev; do
```

(note: this requires changing the README to do the appropriate `kind load` there)

The test VM image isn't required for cluster startup, so I'd rather not require it to exist if we don't need to.

That said, [see note about switching away from these scripts].

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/00-create-vm.yaml:33]: Are `pooler` and/or `metrics` actually used anywhere?

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/00-create-vm.yaml:117]: 512MB here is actually quite big. We might run into issues with VMs with 1GB of memory. Thoughts?

File: tests/e2e/autoscaling/01-upscale.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/01-upscale.yaml:48]: filename should be something other than `my-file.txt`

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/00-create-vm.yaml:5]: At first glance, I feel like this should be handled in `make deploy`, not here. Thoughts? Maybe there's something about kuttl that means this makes more sense?

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/00-create-vm.yaml:86]: Does this work? I'd be concerned about issues around disk migration

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [inline tests/e2e/autoscaling/00-create-vm.yaml:11]: Why do we need a service here?

**Review by sharnoff — COMMENTED** [review]
Looks awesome. Some small items, some questions.

Eventually I'd love to switch away from the scripts directory and only use the makefile — there's some remaining changes required before that happens, though, and I don't think merging this should be blocked by them.

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (bayandin) [reply tests/e2e/autoscaling/00-create-vm.yaml:8]: I've copied this example from `neonvm`. 
Without service it doesn't work: `example:5432 - no response`, I've removed extra stuff that we don't really need (like a bunch of ports and vector), we could add it back if needed (or for other tests)

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (bayandin) [reply tests/e2e/autoscaling/00-create-vm.yaml:59]: It works (as in tests for neonvm). Do you think we need to change it to let's wait until something breaks? 😄

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (bayandin) [reply tests/e2e/autoscaling/00-create-vm.yaml]: Reduced to `256MB` just in case.

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [reply tests/e2e/autoscaling/00-create-vm.yaml:8]: Ahhh I see - didn't realize it was used for providing a stable address. makes sense

File: tests/e2e/autoscaling/00-create-vm.yaml
Comment (sharnoff) [reply tests/e2e/autoscaling/00-create-vm.yaml:59]: Ah ok, yeah should be fine to leave it - plus easy enough to remove later if things break :sweat_smile:

**Review by sharnoff — APPROVED** [review]

**Pull request merged**
